projects/packages/account-protection/tests/php/Password_Manager_Test.php

```diff
@@ -88,24 +93,108 @@ public function test_on_profile_update_with_valid_nonce() {
+
+		$password_manager_mock = $this->create_password_manager_mocks();
+		$password_manager_mock->expects( $this->once() )
+			->method( 'save_recent_password_hash' )
+			->with( $user_id, 'old_password_hash' );
+
+		$password_manager_mock->on_profile_update(
+			$user_id,
+			$old_user_data
+		);
+	}
+
+	public function test_on_profile_update_stdclass_without_user_pass() {
+		$_POST['action'] = 'update';
+
+		$user_id       = 1;
+		$old_user_data = new \stdClass();
+		// Intentionally not setting user_pass
+
+		$password_manager_mock = $this->create_password_manager_mocks();
+		$password_manager_mock->expects( $this->never() )
+			->method( 'save_recent_password_hash' );
+
+		$password_manager_mock->on_profile_update(
+			$user_id,
+			$old_user_data
+		);
+	}
+
+	public function test_on_profile_update_stdclass_with_empty_user_pass() {
+		$_POST['action'] = 'update';
+
+		$user_id                  = 1;
+		$old_user_data            = new \stdClass();
+		$old_user_data->user_pass = '';
+
+		$password_manager_mock = $this->create_password_manager_mocks();
+		$password_manager_mock->expects( $this->never() )
+			->method( 'save_recent_password_hash' );
+
+		$password_manager_mock->on_profile_update(
+			$user_id,
+			$old_user_data
+		);
+	}
+
 	public function test_on_password_reset_saves_recent_password() {
 		$user            = new \WP_User();
 		$user->ID        = 1;
 		$user->user_pass = 'hashedpassword';
 
-		$validation_service_mock = $this->createMock( Validation_Service::class );
-		$password_manager_mock   = $this->getMockBuilder( Password_Manager::class )
-			->setConstructorArgs( array( $validation_service_mock ) )
-			->onlyMethods( array( 'save_recent_password_hash' ) )
-			->getMock();
+		$password_manager_mock = $this->create_password_manager_mocks();
+		$password_manager_mock->expects( $this->once() )
+			->method( 'save_recent_password_hash' )
+			->with( $user->ID, 'hashedpassword' );
 
+		$password_manager_mock->on_password_reset( $user );
+	}
+
+	public function test_on_password_reset_with_stdclass() {
+		$user            = new \stdClass();
+		$user->ID        = 1;
+		$user->user_pass = 'hashedpassword';
+
+		$password_manager_mock = $this->create_password_manager_mocks();
 		$password_manager_mock->expects( $this->once() )
 			->method( 'save_recent_password_hash' )
 			->with( $user->ID, 'hashedpassword' );
 
 		$password_manager_mock->on_password_reset( $user );
 	}
 
+	public function test_on_password_reset_without_user_pass() {
+		$user     = new \stdClass();
+		$user->ID = 1;
+		// Intentionally not setting user_pass
+
+		$password_manager_mock = $this->create_password_manager_mocks();
+		$password_manager_mock->expects( $this->never() )
+			->method( 'save_recent_password_hash' );
+
+		$password_manager_mock->on_password_reset( $user );
+	}
+
+	public function test_on_password_reset_with_empty_user_pass() {
+		$user            = new \stdClass();
+		$user->ID        = 1;
+		$user->user_pass = '';
```

Comment:
Can we add a test case where non-object (say a `null` value) is passed?

Reply:
Done! https://github.com/Automattic/jetpack/pull/43045/commits/176a5160679a76699d257710a0ebfb29ed6953cc